### PR TITLE
Switch Dockerfile.redhat over to use dnf

### PR DESCRIPTION
**Pull request by stevegrubb**
### 🛠 Summary
There is a mix of dnf and yum being used throughout the Dockerfile. This switches it over to only use dnf. As a result, yum-utils is not needed. This also adds more parallelism in downloads and more retries. The nvidia repo sometimes breaks connections. Hopefully the extra retries lets us be successful in setting up the build environment.

### CODE REVIEW

File: Dockerfile.redhat [hunk 1]
```diff
@@ -29,7 +29,8 @@ SHELL ["/bin/bash", "-xo", "pipefail", "-c"]
 ARG JOBS=8
 ARG VERBOSE_LOGS=OFF
 
-RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && dnf clean all && yum update -d6 -y && yum install -d6 -y \
+RUN echo -e "max_parallel_downloads=8\nretries=50" >> /etc/dnf/dnf.conf && \
```
Comment (dtrawins) [inline Dockerfile.redhat:32]:
```suggestion
# hadolint ignore=DL3041
RUN echo -e "max_parallel_downloads=8\nretries=50" >> /etc/dnf/dnf.conf && \
```

File: Dockerfile.redhat [hunk 2]
```diff
@@ -101,19 +101,18 @@ SHELL ["/bin/bash", "-xo", "pipefail", "-c"]
 
 ARG JOBS=40
 
-RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && dnf clean all && yum update -d6 -y && yum install -d6 -y \
+RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && dnf update -d6 -y && dnf install -d6 -y \
```
Comment (dtrawins) [inline Dockerfile.redhat:104]:
```suggestion
# hadolint ignore=DL3041
RUN dnf install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm && dnf update -d6 -y && dnf install -d6 -y \
```

File: Dockerfile.redhat [hunk 3]
```diff
@@ -482,18 +482,17 @@ RUN if [ -f /usr/bin/dnf ] ; then export DNF_TOOL=dnf ; else export DNF_TOOL=mic
         fi ; \
         # For image with Python enabled install Python library
         if ! [[ $debug_bazel_flags == *"PYTHON_DISABLE=1"* ]]; then \
-            $DNF_TOOL install -y python39-libs --setopt=install_weak_deps=0; \
+            $DNF_TOOL install -y python39-libs --setopt=install_weak_deps=0 --nodocs; \
             mkdir -p /ovms/lib/openvino-4.dist-info && \
             echo $'Metadata-Version: 1.0\nName: openvino\nVersion: 2024.4' > /ovms/lib/openvino-2024.4.dist-info/METADATA ; \
         fi ; \
-        $DNF_TOOL install -y shadow-utils; \
```
[the remark this answers is not in the document]
Comment (stevegrubb) [reply Dockerfile.redhat:489]: shadow-utils is in the base image already. So, installing it here is not needed.

File: Dockerfile.redhat [hunk 4]
```diff
@@ -408,8 +408,9 @@ WORKDIR /
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 # hadolint ignore=DL3003,DL3041,SC2164
 RUN if [ -f /usr/bin/dnf ] ; then export DNF_TOOL=dnf ; else export DNF_TOOL=microdnf ; fi ; \
-        $DNF_TOOL upgrade --setopt=install_weak_deps=0 -y ; \
-        $DNF_TOOL install -y pkg-config && rpm -ivh https://vault.centos.org/centos/8/AppStream/x86_64/os/Packages/tbb-2018.2-9.el8.x86_64.rpm && \
+        echo -e "max_parallel_downloads=8\nretries=50" >> /etc/dnf/dnf.conf ; \
```
Comment (dtrawins) [inline Dockerfile.redhat:411]: this doesn't work on ubi-minimal as there is no dnf.conf It make the microdnf unusable

File: Dockerfile.redhat (hunk 3 above)
[the remark this answers is not in the document]
Comment (dtrawins) [reply Dockerfile.redhat:489]: it is missing on ubi-minimal base image

**Comment by dtrawins** [comment]
changes from this PR with resolved conflicts are in https://github.com/openvinotoolkit/model_server/pull/2705

**Pull request closed without merging**